core/src/validator.rs

```diff
@@ -1409,6 +1410,7 @@ impl Validator {
         let cluster_slots =
             Arc::new(crate::cluster_slots_service::cluster_slots::ClusterSlots::default());
 
+        let root_bank_cache = RootBankCache::new(bank_forks.clone());
```

Comment:
refer to https://github.com/anza-xyz/agave/pull/3946/files#r1886312626 for justification of this movement.